File: src/main/java/fr/aven/bot/util/FormatUtils.java
package fr.aven.bot.util;


import java.time.*;
import java.util.stream.Collectors;

public class FormatUtils {

    public static String formatDuration(long millis) {
        return FormatUtils.formatDuration(Duration.ofMillis(millis));
    }

    public static String formatDuration(Duration duration) {
        if (duration.isNegative()) {
            throw new IllegalArgumentException("duration must be positive");
        }
        if (duration.toHours() > 0) {
            return String.format("%d:%02d:%02d", duration.toHoursPart(), duration.toMinutesPart(), duration.toSecondsPart());
        }
        return String.format("%d:%02d", duration.toMinutesPart(), duration.toSecondsPart());
    }

}

File: src/main/kotlin/fr/aven/bot/util/music/FormatUtils.kt
package fr.aven.bot.util.music

import java.time.Duration

class FormatUtils
{
    companion object {
        fun formatDuration(millis: Long): String? {
            return formatDuration(Duration.ofMillis(millis))
        }

        fun formatDuration(duration: Duration): String? {
            require(!duration.isNegative) { "duration must be positive" }
            return if (duration.toHours() > 0) {
                java.lang.String.format(
                    "%d:%02d:%02d",
                    duration.toHoursPart(),
                    duration.toMinutesPart(),
                    duration.toSecondsPart()
                )
            } else java.lang.String.format("%d:%02d", duration.toMinutesPart(), duration.toSecondsPart())
        }
    }
}